### PR TITLE
delete previous image on cover change 

### PR DESCRIPTION
**Pull request by Ahad-patel**
### Feature Preview

Closes: https://github.com/AppFlowy-IO/AppFlowy/issues/5145
 
I have created two methods for deletePreviousImageFromCloudStorage and deleteFile in DocumentSevce Class
when ever the user changes cover image the this method will check if there the previous image was file, If so then it will delete the previous image. 

I have a question ther is a an method DocumentEventDeleteFile which takes DownloadFilePB payload which has two params localFilePath and url, the url can be obtained from coverDetails but localFilepath is not stored any where how do i delete it from the local storage?  @Xazin 

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [x] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [x] All existing tests are passing.

### CODE REVIEW

**Comment by Ahad-patel** [comment]
Hey @LucasXu0, Is my approach to this issue is correct?

File: frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/image/image_util.dart [hunk 1]
```diff
@@ -117,3 +117,7 @@ Future<List<ImageBlockData>> extractAndUploadImages(
 
   return images;
 }
+
+Future<void> deleteImageFromLocalStorage(String localImagePath) async {
+  await File(localImagePath).delete();
```
[the remark this answers is not in the document]
Comment (Ahad-patel) [reply frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/image/image_util.dart]: done

**Comment by Ahad-patel** [comment]
Hey @Xazin, I was trying to add test case in this file => https://github.com/Ahad-patel/AppFlowy/blob/delete-unecessary-images/frontend/appflowy_flutter/integration_test/desktop/document/document_with_cover_image_test.dart#L8-L9, but stuck on how should I approach the local file upload case, Because the file will be present in my machine but might not be availaible in some one else machine, Is there any similar code shoud I look into?

**Comment by Ahad-patel** [comment]
hey @LucasXu0 , @Xazin any updates here?

**Comment by LucasXu0** [comment]
@Ahad-patel In my opinion, the deletion seems unable to be mocked in the test. A workaounrd is to add the flag for the test to detect when the delete function is called while changing the cover image. If the function is called, we can assume the deletion takes effect.

**Comment by Ahad-patel** [comment]
@LucasXu0 , I have added test case for local image upload and added a variable `deleteImageTestCounter` to test if `deleteImageFromLocalStorage` is being called when we remove/change cover image.

**Review by LucasXu0 — APPROVED** [review]

**Pull request merged**